janitor/functions.py

```diff
@@ -3869,3 +3876,104 @@ def sort_naturally(
     """
     new_order = index_natsorted(df[column_name], **natsorted_kwargs)
     return df.iloc[new_order, :]
+
+
+@pf.register_dataframe_method
+def expand_grid(
+    df: pd.DataFrame = None, df_key: str = None, others: Dict = None
+) -> pd.DataFrame:
+    """ Creates a dataframe from a combination of all inputs.
+        This works with a dictionary of name value pairs, 
+        and will work with structures that are not dataframes.
+        If method-chaining to a dataframe, 
+        a key to represent the column name in the output must be provided.
+        The output will always be a dataframe.
```

Comment:
The indentation on the docstring formatting is incorrect here. Suggestion below:

```suggestion
    """ 
    Create a dataframe from a combination of all inputs.
    
    This works with a dictionary of name value pairs, 
    and will work with structures that are not dataframes.
    If method-chaining to a dataframe, 
    a key to represent the column name in the output must be provided.
    
    The output will always be a dataframe.
```